common/k8s/src/middleware/metadata.rs

```diff
@@ -231,10 +231,31 @@ impl K8sMetadata {
             ListParams::default()
         };
 
-        let watcher = Box::pin(StreamBackoff::new(
-            watcher(api, params),
-            ExponentialBackoff::default(),
-        ));
+        let watcher = watcher(api, params).map_ok(|ev| {
+            ev.modify(|pod| {
+                pod.managed_fields_mut().clear();
```

Comment:
Clearing managed_fields happens twice, once here and again on L384. You can remove the `map_ok` on L384